Let Firefox on Android manage all storage types.

mlindsay opened the pull request:
Firefox on Android now supports more than just cookies so the extension should allow users to manage those as well.

Replaces many uses of isFirefoxNotAndroid with simply isFirefox, but leaves in place some that are mainly about UI limitations. Completely removes the isFirefoxAndroid helper.

Without this many people may not realize that this extension is letting local storage persist through cookie cleaning, especially if they are familiar with the extension on desktop.

kennethtran93 commented:
Instead of fully removing it, it needs to check if the extension is installed in version 85+ for Firefox Android, since earlier versions do not have these capabilities.

mlindsay commented:
Hey Kenneth,

Just wanted to verify that (Firefox && android && ffVersion >= 85) is definitely what you want before I try to update the change. In part because the capability was added in Firefox 58 and I'm not sure if you simply typo'd that. I understand release 85 is when Firefox Android started allowing most normal extensions back onto it, but I don't think it is when the capabilities were actually added. So overall I don't think there was ever a path for Firefox Android to load this extension without using developer escape hatches prior to 85, so it can't really be broken on those older versions.

Thanks

mlindsay committed:
Let Firefox on Android manage all storage types.

Firefox on Android now supports more than just cookies so the extension
should allow users to manage those as well.

Adds a method to generate a map of the current supported storage types
so that they can be easily queried. This isolates all the version
checking to one place as well.

mlindsay commented:
Thanks for the link and the explanation. I've updated the change to follow with that chart. Glad for your testing as well, I made a mistake in transcribing the version numbers in one place and it triggered one of tests to fail.
